Doc/glossary.rst

```diff
@@ -1011,6 +1095,16 @@ Glossary
       properties, :meth:`~object.__getattribute__`, class methods, and static
       methods.
 
+   non-deterministic
+      Behavior where the outcome of a program can vary between executions with
+      the same inputs.  In multi-threaded programs, non-deterministic behavior
+      often results from :term:`race conditions <race condition>` where the
+      relative timing or interleaving of threads affects the result.
+      :term:`Data races <data race>` are a common cause of non-deterministic
+      bugs.  Proper synchronization using :term:`locks <lock>` and other
```

Comment:
```suggestion
      the same inputs.  An example of non-deterministic behavior
      are :term:`race conditions <race condition>`.
      Proper synchronization using :term:`locks <lock>` and other
```

Reply:
I just removed the sentence @ngoldbaum mentioned. Does it read okay like this? I'd like to keep the mention of `relative timing` and `interleaving` in.